docs/modules/ROOT/pages/how-tos/configuring/adding-custom-tasks.adoc

```diff
@@ -0,0 +1,146 @@
+= Adding custom tasks
+
+If your pipeline uses xref:advanced-how-tos/using-trusted-artifacts.adoc[Trusted Artifacts],
+you can add custom tasks to the pipeline without compromising the chain of trust.
+These tasks can do just about anything you'd like: set dynamic arguments for the build, run unit tests, send emails, etc.
+With one major restriction: they *cannot modify your source repo or the prefetched content* before the build.
+Doing so would violate the link:https://enterprisecontract.dev/docs/ec-policies/release_policy.html#trusted_task__trusted[Trusted Tasks] Enterprise Contract rule.
+
+What this means in Trusted Artifacts terms:
+
+* To operate on your repo, your Task must declare a `SOURCE_ARTIFACT` parameter and use a special step to download this artifact
+  (refer to the ready-to-use examples below).
+* Your Task must *not* return a modified `SOURCE_ARTIFACT`.
+  It's ok if the task modifies the source repo as it executes, but the modifications will be discarded.
+  The build will use the original, unmodified `SOURCE_ARTIFACT`.
++
+NOTE: If you do return a modified `SOURCE_ARTIFACT` from a custom task and pass it to the build task,
+    Enterprise Contract will detect that the chain of trust contains an untrusted task and report the violation.
+
+.Prerequisites:
+
+. Your pipeline uses xref:advanced-how-tos/using-trusted-artifacts.adoc[Trusted Artifacts].
+
+.Procedure:
+
+. Implement the custom Task in the `.tekton/tasks/` directory in your repo.
++
+NOTE: The presence of the Task in `.tekton/**` is enough to make it available for use in your Pipelines without any extra configuration;
+    see link:https://pipelinesascode.com/docs/guide/resolver/[Pipelines-as-Code resolver].
+. Use the Task in your Pipelines as desired.
+
+Useful resources for writing Tekton tasks:
+
+* link:https://tekton.dev/docs/pipelines/tasks/[Tekton Task docs]
+* link:https://github.com/konflux-ci/build-definitions/tree/main/task[Existing Konflux tasks]
+
+== Examples
+
+=== Dynamically set the ARGs and LABELs for the container build
+
+Implement the task in e.g. `.tekton/tasks/get-buildah-args.yaml`:
+
+[source,yaml]
```

Comment:
We have a repository where we can create some sample pipelines/tasks: https://github.com/konflux-ci/pipeline-samples

What do you think about propose this task there instead so that readers won't have to extract it out of the documentation?